src/when.js

```diff
@@ -82,11 +83,11 @@ class WhenMock {
           return a.id - b.id
         })
 
-      this.nextCallMockId++
+      thus.nextCallMockId++
 
-      this.fn.mockImplementation((...args) => {
-        for (let i = 0; i < this.callMocks.length; i++) {
-          const { matchers, mockImplementation, expectCall, once, called } = this.callMocks[i]
+      this.fn.mockImplementation(function (...args) {
```

Comment:
I think only this function needs the `thus` stuff, all the above `thus` I think can be reverted back to `this`. Cause right here is the only place we use the `function` declaration instead of the arrow function.

Also can we name `thus` to `instance`? Otherwise looking good thanks for the tests!

Reply:
Good point.
Fixed. Rebased.